### PR TITLE
New integrator, and add some metadata to integrators.py

### PR DESCRIPTION
**Pull request by reubenharry**
Addresses #679 

 - [x] We should be able to understand what the PR does from its title only;
 - [x] There is a high-level description of the changes;
 - [x] There are links to *all* the relevant issues, discussions and PRs;
 - [x] The branch is rebased on the latest `main` commit;
 - [x] Commit messages follow these [guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html);
 - [x] The code respects the current naming conventions;
 - [x] Docstrings follow the [numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html)
 - [x] `pre-commit` is installed and configured on your machine, and you ran it before opening the PR;
 - [x] There are tests covering the changes;
 - [x] The doc is up-to-date;

### CODE REVIEW

**Comment by reubenharry** [comment]
I see that code coverage is complaining about `integrator_order`, `name_integrator` and `calls_per_integrator_step`. These seem useful, but depending on your view, I could remove them

**Comment by junpenglao** [comment]
> I see that code coverage is complaining about `integrator_order`, `name_integrator` and `calls_per_integrator_step`. These seem useful, but depending on your view, I could remove them

Yeah please remove them, depending on what you want to check there should be better ways to return these properties.

File: blackjax/mcmc/integrators.py [hunk 1]
```diff
@@ -23,13 +23,23 @@
 from blackjax.types import ArrayTree
 
 __all__ = [
+    "velocity_verlet_coefficients",
+    "mclachlan_coefficients",
+    "yoshida_coefficients",
+    "omelyan_coefficients",
```
Comment (junpenglao) [inline blackjax/mcmc/integrators.py:29]: Remove? Are you planning to use it outside of integrators?

**Review by junpenglao — CHANGES_REQUESTED** [review]

File: blackjax/mcmc/integrators.py (hunk 1 above)
Comment (reubenharry) [reply blackjax/mcmc/integrators.py:29]: Also, it seems like code duplication to have `isokinetic_velocity_verlet`, `velocity_verlet`, etc, when we could just use the coefficients with `generate_isokinetic_integrator` and `generate_euclidean_integrator`

Comment (junpenglao) [reply blackjax/mcmc/integrators.py:29]: > Also, it seems like code duplication to have isokinetic_velocity_verlet, velocity_verlet, etc, when we could just use the coefficients with generate_isokinetic_integrator and generate_euclidean_integrator

The design choice is motivated by how we are envisioning the usage of integrators in the library. Currently the design is to have integrator being a static set.

> As another example, I often want to do benchmarks against different integrators, but want to just iterate over X_coefficients, and then use generate_isokinetic_integrator for MCLMC and generate_euclidean_integrator for HMC

You should iterate through the integrator objects instead

> For example, if I want to run scripts that try different integrators, and I want access to their number of gradient calls. I suppose the other option would be to have a dictionary like {"velocity_verlet": {"num_grads": ..., "name": ..., "order": ...} }. Is that what you'd recommend?

Yes given that these are static, you should put them in your script as static parameters. I dont yet see those are useful in the library outside of benchmarking.

Comment (reubenharry) [reply blackjax/mcmc/integrators.py:29]: OK, that makes sense. I think my one point of disagreement is that nothing in the code "knows" that `velocity_verlet` and `isokinetic_velocity_verlet` are related. So I will have to have a dictionary of `{"euclidean": velocity_verlet, "isokinetic": isokinetic_velocity_verlet, ...}` when I want to compare each integrator on hmc vs mclmc, which I'm currently doing. This is a little painful, but not the end of the world

Comment (reubenharry) [reply blackjax/mcmc/integrators.py]: Updated, as per request

**Review by junpenglao — APPROVED** [review]
